### ISSUE TITLE
[BUG]: NaN issue again after commit e67b4f2

### ISSUE THREAD

**Issue by Fangzhou-Ai** [vllm-project/vllm#1134]
It seems that the old issue #936 happened again after this PR #1004. An easy test could be
```python
from vllm import LLM, SamplingParams
import contextlib
import torch
from transformers import AutoModelForCausalLM, AutoTokenizer
from typing import List

def vllm(prompt : str):
  with contextlib.nullcontext():
    llm = LLM("lmsys/vicuna-7b-v1.3")
    sampling_params = SamplingParams(max_tokens=128, temperature=0) # Greedy sampling
    print(prompt, llm.generate(prompt, sampling_params)[0].outputs[0].text)


if __name__ == "__main__":
  prompt = "San Francisco is a city that"
  vllm(prompt)
```
The output is wrong.
After reverting the commit e67b4f2 the output is corect

**Comment by viktor-ferenczi** [comment]
@WoosukKwon This is something more complex, which needs expert attention.

**Comment by WoosukKwon** [comment]
@Fangzhou-Ai Thanks for reporting the issue. It's actually not an NaN error, but it's due to numerical differences between our custom CUDA kernels and the original HF implementation. These numerical differences are inevitable because floating-point arithmetics are not associative.

While the numerical differences can lead to drastically different outputs for some examples, we've that it does not hurt the task accuracy (e.g., ARC and Hellaswag).
